src/actors/cficaches.rs

```diff
@@ -220,32 +225,31 @@ impl CfiCacheActor {
         let identifier = request.identifier.clone();
         let scope = request.scope.clone();
 
-        object.boxed_local().compat().and_then(move |object| {
-            object
-                .meta
-                .map(move |object_meta| {
-                    Either::A(cficaches.compute_memoized(FetchCfiCacheInternal {
-                        request,
-                        objects_actor: objects,
-                        object_meta,
-                        threadpool,
-                    }))
-                })
-                .unwrap_or_else(move || {
-                    Either::B(
-                        Ok(Arc::new(CfiCacheFile {
+        object
+            .and_then(move |object| {
```

Comment:
Consider this optional for this PR, but I would like to have this addressed in follow-ups: 

This function is large and uses unidiomatic combinators where async/await could be used. We could rewrite this into an `async fn`, split it up in smaller parts with speaking names, and then call a single `async fn` in the trait impl, surrounded by `Box::pin`.

Reply:
Yes, this is definitely something that should happen, but definitely as a followup.